pulser-core/pulser/sequence/sequence.py

```diff
@@ -1338,14 +1337,17 @@ def draw(
             draw_modulation="output" in mode,
             draw_phase_curve=draw_phase_curve,
         )
-        if fig_name is not None and draw_register:
+        if fig_name is not None and fig_reg is not None:
             name, ext = os.path.splitext(fig_name)
             fig.savefig(name + "_pulses" + ext, **kwargs_savefig)
             fig_reg.savefig(name + "_register" + ext, **kwargs_savefig)
         elif fig_name:
             fig.savefig(fig_name, **kwargs_savefig)
```

Comment:
Is there a reason why we don't always have the same name for the figure files ? `fig.savefig(name + "_pulses" + ext, **kwargs_savefig)` ?

Reply:
Well, we figured there was no need to change the name of the figure provided by the user unless we needed to distinguish between the register and the pulse sequence.